engine/src/main.cpp

```diff
@@ -760,139 +761,134 @@ void bootstrap_main_loop()
-                    //don't ignore the network section of the config file
-                    ignore_network = false;
-                } else if (strcmp( lpCmdLine[i], "--help" ) == 0) {
-                    cout<<helpmessage;
-                    exit( 0 );
-                } else if (strcmp( lpCmdLine[i], "--version" ) == 0) {
-                    cout<<versionmessage;
-                    exit( 0 );
-                } else if (strncmp( lpCmdLine[i], "--debug", 7 ) == 0) {
-                    if (lpCmdLine[i][7] == 0) {
-                        g_game.vsdebug = 1;
-                    } else if (lpCmdLine[i][8] == 0) {
-                        cout<<helpmessage;
-                        exit( 0 );
-                    }
-                    g_game.vsdebug = lpCmdLine[i][8] - '0';
-                    ++i;
-                }
-                break;
-            }
+    QVector PlayerLocation;
+    boost::program_options::options_description vs_switches("Command line options for vegastrike; short options are case insensitive");
+    vs_switches.add_options()
+        (",d", boost::program_options::value<std::string>(), "Specify data directory, full path expected")
+        (",n", boost::program_options::value<int>(), "Number of players")
+        (",m", boost::program_options::value<std::string>(), "Specify a mod to play")
+        // (",p", boost::program_options::value<std::string>(), "Specify player location")  // wasn't handled, don't know the exact format
+        (",j", boost::program_options::value<std::string>(), "Start in a specific system")
+        (",l", boost::program_options::value<std::string>(), "Specify player location")
+        (",a", "Normal resolution (800x600)")
+        (",h", "High resolution (1024x768)")
+        (",v", "Super high resolution (1280x1024)")
+        ("help", "Show this help")
+        ("benchmark", boost::program_options::value<float>(), "???")
+        ("net", "Networking Enabled (Experimental)")
+        ("debug", boost::program_options::value<int>()->implicit_value(1), "Enable debugging output, 1 major warnings, 2 medium, 3 developer notes")
+        ("test-audio", "Run audio tests")  // is handled in readCommandLineOptions, here is serves only for help message
+        ("version", "Print the version and exit")
+        ;
+
+    boost::program_options::options_description vs_modules;
+    vs_modules.add_options()
+        ("mission_name", boost::program_options::value<std::string>(), "Mission name")
+        ;
+
+    boost::program_options::options_description vs_options;
+    vs_options.add(vs_switches).add(vs_modules);
+
+    boost::program_options::positional_options_description vs_positionals;
+    vs_positionals.add("mission_name", 1);
+
+    boost::program_options::command_line_style::style_t style = boost::program_options::command_line_style::style_t(
+        boost::program_options::command_line_style::unix_style |
+        boost::program_options::command_line_style::case_insensitive
+    );
```

Comment:
:+1: Awesome.
We might relax from `unix_style` later so it matches each platform properly, but this is good for transition to the new support.

Reply:
Actually, I hate all the code around checking which option was passed :) Too much code for very little action.

Yes, windows users might be happier with `/` instead of `-`. This might be easily changed with some `#ifdef` or something like that.

Reply:
yeah - but one step at a time. I could see us moving this over to the main function at some point so that we don't get this far into the code if we can't pass the options. Regardless this is a good first step. thanks!

Reply:
also for 0.8.x we need to maintain full compatibility - no changes like swapping parameter styles. for 0.9.x (current master) we could break that compatibility and allow the platforms to use their native parameter styles.

Reply:
> for 0.9.x (current master) we could break that compatibility and allow the platforms to use their native parameter styles.

Should I create new Issue for this?

Reply:
@ondrejkajinek Sure, feel free to do so. You can create an issue directly off any of the above comments.

Reply:
Here: https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/775